language/diem-framework/modules/0L/FullnodeSubsidy.move

```diff
@@ -17,7 +17,8 @@ address 0x1 {
     use 0x1::TowerState;
 
     public fun get_proof_price(one_val_subsidy: u64): u64 {
-      let global_proofs = TowerState::get_fullnode_proofs();
+
```

Comment:
I still have a concern. Here we calculate the price we pay per proof as the (total_reward / number_of_proofs_mined_above_threshold), but in payouts, we don't pay miners only for their proofs above threshold, we pay them for all their proofs (assuming they mined over threshold). This will result in the total reward paid out being much higher then intended. In the worst case, all miners mine 7 proofs in the epoch, in which case 7x the intended reward will be paid out. 

We either need to 
- only pay miners for proofs above threshold
- calculate the payout per proof as (total_reward / number_of_proofs_mined), though this will result in underpayment assuming some miner(s) mine below threshold
- calculate the payout per proof as (total_reward / number_of_proofs_mined_by_miners_above_threshold) (this requires a different calculation here)

Reply:
So to be clear the way this is implemented is your suggestion #1. We are only paying for proofs subsequent to the threshold being met. I think it's conceptually correct, and doing anything else is tricky. Looking ahead "pay miners for proofs above threshold" also works better for lazy computation, instead of having to loop through all miners an evaluate.

Reply:
Great, the change to EpochBoundary resolves this concern.